cluster/addons/dns/kube2sky/kube2sky.go

```diff
@@ -100,26 +123,38 @@ type kube2sky struct {
 	podsStore kcache.Store
 	// Lock for controlling access to headless services.
 	mlock sync.Mutex
+	// If true, perform quorum read.
+	quorum bool
 }
 
 // Removes 'subdomain' from etcd.
 func (ks *kube2sky) removeDNS(subdomain string) error {
 	glog.V(2).Infof("Removing %s from DNS", subdomain)
-	resp, err := ks.etcdClient.RawGet(skymsg.Path(subdomain), false, true)
+	ctx := context.TODO()
+	opts := &etcd.GetOptions{
+		Recursive: true,
+		Quorum:    ks.quorum,
+	}
+	resp, err := ks.etcdKeysAPI.Get(ctx, skymsg.Path(subdomain), opts)
```

Comment:
We could probably reduce the block to just the delete.  The read check is unnecessary imo.